day2/benches/bench.rs

```diff
@@ -0,0 +1,10 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let input = include_str!("../input.txt");
+    c.bench_function("part1", |b| b.iter(|| day2::part1(black_box(input))));
```

Comment:
From last time, my understanding of `black_box()` is that it prevents the super-smart rust compiler from optimizing stuff away.  I haven't built up a good intuition for when that's needed, but I'm surprised that it's put around `input` here, since `input` is just a value which doesn't need to be computed at this point.  Can you help me understand the reasoning here?

Reply:
Great question! I probably can't because I have no idea what I'm talking about. But, to give the appearance of wisdom (which will backfire and reveal my ignorance), I'll type a bunch of words and hope you don't read it all and just assume I know what I'm talking about.

My rationale here was that `input` is known at compile time to this function (because it's imported with that macro). That means the compiler might look ahead at it and determine weird stuff when it comes to what gets passed to `part1()` (which is fairly simple). By `black_box`ing `input` _before_ it gets passed to `part1()` we try to ensure:
1. the compiler doesn't assume it's getting the same input as the last iteration and so just copying the result from the previous iteration
2. the compiler doesn't look ahead at the value and precompute anything

That being said it is definitely guesswork on my end! My vague knee-jerk reaction after reading [that blog post](https://gendignoux.com/blog/2022/01/31/rust-benchmarks.html) is to always `black_box` the input and output (the output here is done by the `criterion` library). That being said I didn't do that in the Rust BST PR (assuming BSTs were way too complicated for the compiler to precompute anything on) and the right way to do this would be to go to a compiler explorer and see what the generated code is with and without `black_box` so we measure only _exactly_ what we care about. But I'm too stupid/lazy to do that here :speak_no_evil: 

Glad to spend the time in the compiler explorer together if we want to because that'd be way more fun together!